includes/wc-product-functions.php

```diff
@@ -494,49 +459,35 @@ function wc_get_formatted_variation( $variation, $flat = false ) {
 
 /**
  * Function which handles the start and end of scheduled sales via cron.
- *
- * @access public
  */
 function wc_scheduled_sales() {
-	global $wpdb;
+	$data_store = WC_Data_Store::load( 'product' );
 
 	// Sales which are due to start
-	$product_ids = $wpdb->get_col( $wpdb->prepare( "
-		SELECT postmeta.post_id FROM {$wpdb->postmeta} as postmeta
-		LEFT JOIN {$wpdb->postmeta} as postmeta_2 ON postmeta.post_id = postmeta_2.post_id
-		LEFT JOIN {$wpdb->postmeta} as postmeta_3 ON postmeta.post_id = postmeta_3.post_id
-		WHERE postmeta.meta_key = '_sale_price_dates_from'
-		AND postmeta_2.meta_key = '_price'
-		AND postmeta_3.meta_key = '_sale_price'
-		AND postmeta.meta_value > 0
-		AND postmeta.meta_value < %s
-		AND postmeta_2.meta_value != postmeta_3.meta_value
-	", current_time( 'timestamp' ) ) );
-
+	$product_ids = $data_store->get_starting_sales();
 	if ( $product_ids ) {
 		foreach ( $product_ids as $product_id ) {
-			$sale_price = get_post_meta( $product_id, '_sale_price', true );
+			$product = wc_get_product( $product_id );
+			$sale_price = $product->get_sale_price();
 
 			if ( $sale_price ) {
-				update_post_meta( $product_id, '_price', $sale_price );
+				$product->set_price( $sale_price );
 			} else {
-				// No sale price!
-				update_post_meta( $product_id, '_sale_price_dates_from', '' );
-				update_post_meta( $product_id, '_sale_price_dates_to', '' );
+				$product->set_date_on_sale_to( '' );
+				$product->set_date_on_sale_from( '' );
 			}
 
-			$parent = wp_get_post_parent_id( $product_id );
+			$product->save();
 
+			$parent = $product->get_parent_id();
 			// Sync parent
 			if ( $parent ) {
 				// Clear prices transient for variable products.
 				delete_transient( 'wc_var_prices_' . $parent );
 
 				// Grouped products need syncing via a function
-				$this_product = wc_get_product( $product_id );
-
-				if ( $this_product->is_type( 'simple' ) ) {
-					$this_product->grouped_product_sync();
+				if ( $product->is_type( 'simple' ) ) {
```

Comment:
Just a note, I need to go over the grouped sync at some point and review if needed.